rmw_fastrtps_cpp/src/rmw_client.cpp

```diff
@@ -197,9 +197,9 @@ rmw_create_client(
       delete info->pub_listener_;
       delete info->listener_;
       rmw_fastrtps_shared_cpp::remove_topic_and_type(
-        participant_info, info->response_topic_, info->response_type_support_);
+        participant_info, nullptr, info->response_topic_, info->response_type_support_);
       rmw_fastrtps_shared_cpp::remove_topic_and_type(
-        participant_info, info->request_topic_, info->request_type_support_);
+        participant_info, nullptr, info->request_topic_, info->request_type_support_);
```

Comment:
For consistency with publishers and subscriptions, I would move the calls to `remove_topic_and_type` before the deletion of listeners.

If this is taken, applies also to services, and also to the dynamic RMW

Reply:
Good call.  Fixed in 222d3ce43fffe6191a7ee544bb21bce915f688e1.